spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/logging/LoggingApplicationListener.java

```diff
@@ -303,7 +303,7 @@ private void initializeEarlyLoggingLevel(ConfigurableEnvironment environment) {
 
 	private boolean isSet(ConfigurableEnvironment environment, String property) {
 		String value = environment.getProperty(property);
-		return (value != null && !value.equals("false"));
+		return (value != null && !"false".equals(value));
```

Comment:
Can be replaced by `Boolean.parseBoolean`

Reply:
!"false".equals(value) not equals s.equalsIgnoreCase("true"),eg.If enter an empty string, the former return true but the latter return false.

Reply:
Yeah, you're right there. Maybe not the best idea. I'll make up my mind as part of polishing this.